lib/galaxy/datatypes/registry.py

```diff
@@ -1054,9 +1054,8 @@ def find_conversion_destination_for_dataset_by_extensions(
             ext = dataset_or_ext
             dataset = None
 
-        if self.get_datatype_by_extension(ext) is not None and self.get_datatype_by_extension(ext).matches_any(
-            accepted_formats
-        ):
+        datatype_by_extension = self.get_datatype_by_extension(ext)
+        if datatype_by_extension is not None and datatype_by_extension.matches_any(accepted_formats):
```

Comment:
```suggestion
        if datatype_by_extension and datatype_by_extension.matches_any(accepted_formats):
```

it's a dict lookup, I doubt this makes a practical difference in performance, but of course it's good to do it just once.